storage/innobase/fil/fil0fil.cc

```diff
@@ -2084,6 +2093,22 @@ fil_rename_tablespace_check(
 
 	exists = false;
 	if (os_file_status(new_path, &exists, &ftype) && !exists) {
+		auto schema_path= fil_make_filepath(new_path, NULL, NO_EXT,
+						    true, true);
+		if (schema_path == NULL) {
+			return DB_ERROR;
+		}
+		exists= false;
+		if (os_file_status(schema_path, &exists, &ftype) && !exists)
+		{
+			ib::error() << "Cannot rename '" << old_path
+				    << "' to '" << new_path
+				    << "' because the target schema directory"
+				    << " doesn't exist.";
+			ut_free(schema_path);
+			return DB_ERROR;
+		}
+		ut_free(schema_path);
 		return DB_SUCCESS;
```

Comment:
We could assign the return value to a variable, to avoid code duplication:
```c++
ut_free(schema_path);
return err;
```

Reply:
Now restructured the code a bit based on other comments. Please check. ut_free is done in 2 places but should be ok I think.